data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3SourceConfig.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.plugins.source.configuration.AWSAuthenticationOptions;
+import com.amazon.dataprepper.plugins.source.configuration.CodecOption;
+import com.amazon.dataprepper.plugins.source.configuration.CompressionOption;
+import com.amazon.dataprepper.plugins.source.configuration.NotificationTypeOption;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
+
+public class S3SourceConfig {
+
+    @JsonProperty("notification_type")
+    private NotificationTypeOption notificationType;
```

Comment:
This should have a NotNull annotation. Then you can entirely remove the `isNotificationTypeValid()` method.